src/logic/reminders.ts

```diff
@@ -0,0 +1,82 @@
+import { isTest } from '../utils/env';
+
+const REMINDER_ENDPOINT_PROD = 'https://support.theguardian.com/reminders/create/one-off';
+// const REMINDER_ENDPOINT_CODE = 'https://support.code.dev-theguardian.com/reminders/create/one-off';
+
+export type ReminderStatus = 'DEFAULT' | 'IN_PROGRESS' | 'SUCCESS' | 'FAILURE';
+
+export type ReminderPlatform = 'WEB' | 'AMP';
+
+export type ReminderComponent = 'EPIC' | 'BANNER';
+
+// The oneoff reminder will only be set for unrecognised emails if reminderStage === PRE
+// Not an issue in Prod because only signed in users will see the banner or epic
+// - but can cause heartbreak while testing locally with an unrecognised email address
+export type ReminderStage = 'PRE' | 'POST' | 'WINBACK';
+
+export interface BaseSignupRequest {
+    email: string;
+    reminderPlatform: ReminderPlatform;
+    reminderComponent: ReminderComponent;
+    reminderStage: ReminderStage;
+    country?: string;
+    reminderOption?: string;
+}
+
+export interface OneOffSignupRequest extends BaseSignupRequest {
+    reminderPeriod: string;
+}
+
+const getReminderDate = (date: Date): Date => {
+    const monthsAhead = date.getDate() < 20 ? 1 : 2;
+    date.setMonth(date.getMonth() + monthsAhead);
+
+    return date;
+};
+
+export interface ReminderFields {
+    reminderCta: string;
+    reminderLabel: string;
+    reminderPeriod: string;
+    reminderOption?: string;
+}
+
+export const buildReminderFields = (today: Date = new Date()): ReminderFields => {
+    const reminderDate = getReminderDate(today);
+
+    const month = reminderDate.getMonth() + 1; // javascript dates run from 0-11, we want 1-12
+    const paddedMonth = month.toString().padStart(2, '0');
+    const monthName = reminderDate.toLocaleString('default', { month: 'long' });
+    const year = reminderDate.getFullYear();
+
+    return {
+        reminderCta: `Remind me in ${monthName}`,
+        reminderPeriod: `${year}-${paddedMonth}-01`,
+        reminderLabel: `${monthName} ${year}`,
+    };
+};
+
+export const createReminder = (signupData: OneOffSignupRequest): Promise<void> => {
+    const url = REMINDER_ENDPOINT_PROD;
```

Comment:
Could we use `REMINDER_ENDPOINT_PROD` directly rather than assign to a const?

Reply:
Yes - this is a result of an attempt earlier in the history of this PR to differentiate endpoints based on code environment - but as RRCP only uses PROD for reminders I was persuaded to abandon that effort. Will update.

Reply:
Fixed